### ISSUE TITLE
Correct linting error context in agentskills:edit_file

### ISSUE THREAD

**Issue by li-boxuan** [OpenDevin/OpenDevin#2210]
Looks like we could potentially improve the performance on python benchmarks (swe-bench for example) if we could fix this.

In the below example, the LLM issues an edit from line 9 to line 23, and linter catches an error. Our agentskills toolbox shows the updated file but starts from line 13, even though the linter says the error is from line 9.

Ideally, we should parse/capture the error line number from the linter, and show the context around that line, rather than the middle of the edit (current logic)

```
I have fixed the indentation issue in the `check` function. Let me update the file for you:
CODE:
edit_file(start=9, end=23, content="def check(any_int):\n    # Check some simple cases\n    assert any_int(2, 3, 1)==True, \"This prints if this assert fails 1 (good for debugging!)\"\n    assert any_int(2.5, 2, 3)==False, \"This prints if this assert fails 2 (good for debugging!)\"\n    assert any_int(1.5, 5, 3.5)==False, \"This prints if this assert fails 3 (good for debugging!)\"\n    assert any_int(2, 6, 2)==False, \"This prints if this assert fails 4 (good for debugging!)\"\n    assert any_int(4, 2, 2)==True, \"This prints if this assert fails 5 (good for debugging!)\"\n    assert any_int(2.2, 2.2, 2.2)==False, \"This prints if this assert fails 6 (good for debugging!)\"\n    assert any_int(-4, 6, 2)==True, \"This prints if this assert fails 7 (good for debugging!)\"\n\n    # Check some edge cases that are easy to work out by hand.\n    assert any_int(2,1,1)==True, \"This prints if this assert fails 8 (also good for debugging!)\"\n    assert any_int(3,4,7)==True, \"This prints if this assert fails 9 (also good for debugging!)\"\n    assert any_int(3.0,4,7)==False, \"This prints if this assert fails 10 (also good for debugging!)\"\n\ncheck(any_int)")
16:15:54 - OBSERVATION
**IPythonRunCellObservation**
[Your proposed edit has introduced new syntax error(s). Please understand the errors and retry your edit command.]
ERRORS:
/workspace/Python__92.py:9:2: E999 IndentationError: expected an indented block after function definition on line 7
[This is how your edit would have looked if applied]
-------------------------------------------------
(12 more lines above)
13|    assert any_int(1.5, 5, 3.5)==False, "This prints if this assert fails 3 (good for debugging!)"
14|    assert any_int(2, 6, 2)==False, "This prints if this assert fails 4 (good for debugging!)"
15|    assert any_int(4, 2, 2)==True, "This prints if this assert fails 5 (good for debugging!)"
16|    assert any_int(2.2, 2.2, 2.2)==False, "This prints if this assert fails 6 (good for debugging!)"
17|    assert any_int(-4, 6, 2)==True, "This prints if this assert fails 7 (good for debugging!)"
18|
19|    # Check some edge cases that are easy to work out by hand.
20|    assert any_int(2,1,1)==True, "This prints if this assert fails 8 (also good for debugging!)"
21|    assert any_int(3,4,7)==True, "This prints if this assert fails 9 (also good for debugging!)"
22|    assert any_int(3.0,4,7)==False, "This prints if this assert fails 10 (also good for debugging!)"
(2 more lines below)
-------------------------------------------------

[This is the original code before your edit]
-------------------------------------------------
(12 more lines above)
13|    assert any_int(2, 6, 2)==False, "This prints if this assert fails 4 (good for debugging!)"
14|    assert any_int(4, 2, 2)==True, "This prints if this assert fails 5 (good for debugging!)"
15|    assert any_int(2.2, 2.2, 2.2)==False, "This prints if this assert fails 6 (good for debugging!)"
16|    assert any_int(-4, 6, 2)==True, "This prints if this assert fails 7 (good for debugging!)"
17|
18|    # Check some edge cases that are easy to work out by hand.
19|    assert any_int(2,1,1)==True, "This prints if this assert fails 8 (also good for debugging!)"
20|    assert any_int(3,4,7)==True, "This prints if this assert fails 9 (also good for debugging!)"
21|    assert any_int(3.0,4,7)==False, "This prints if this assert fails 10 (also good for debugging!)"
22|
(1 more lines below)
-------------------------------------------------
Your changes have NOT been applied. Please fix your edit command and try again.
You either need to 1) Specify the correct start/end line arguments or 2) Correct your edit code.
DO NOT re-run the same failed edit command. Running it again will lead to the same error.
```

Acceptance criteria:
1. Add a unit test that can reproduce the problem
2. Bump CodeActAgent and CodeActSWEAgent versions

Again, we shouldn't merge the fix before we finish the current round of evaluations on CodeActAgent/CodeActSWEAgent 1.5.

**Comment by tobitege** [comment]
I've add an extra test *trying* to reproduce the issue you experienced, but I think the issue is no longer.

https://github.com/OpenDevin/OpenDevin/pull/2307/commits/2d8ddf2756aca4153cceba805ee783c10440ff8e

**Comment by tobitege** [comment]
This issue should be fixed when #2346 is merged:
that PR has some tweaked lint message processing in it to get the first error line number which should then be used as the current line to display the file.

**Comment by mamoodi** [comment]
Going to close based on tobitege's response and PR merging. Please reopen if this is not the case.
